### DISCUSSION TITLE
Winget force closing app when kicking off an upgrade - By design?

### DISCUSSION THREAD

**Discussion by romnation1** [microsoft/winget-cli#4668] [General]
I am using winget in an organization in combination with this auto updater here:

https://github.com/Romanitho/Winget-AutoUpdate

One thing i noticed is manually triggering winget to update a package via powershell force closes the app. Is this the same experience my users are going to experience? I do NOT want it to force close any running apps, especially if this is something we are going to have kick off daily or at logon, as we do not want to interrupt the user.

**Comment by Masamune3210** [comment]
If I'm not mistaken, its actually the installers themselves that are force closing their respective programs. Most installers by default will either force close the program before performing operations on its files or will outright refuse to proceed if the program is detected as being open.

**Reply by Trenly** [reply]
I believe this is true. At its core, all WinGet does is download the installer and run it with the necessary switches to perform an upgrade. 

I suppose in some cases it could be that the manifest specifies a switch to force close the application. If that is the case then that particular switch should be removed from the manifest and `ExpectedReturnCodes` should be used instead
